src/chatterbox/models/voice_encoder/voice_encoder.py

```
@@ -188,10 +188,10 @@ def inference(self, mels: torch.Tensor, mel_lens, overlap=0.5, rate: float=None,
 
         # Forward the partials
         n_chunks = int(np.ceil(len(partials) / (batch_size or len(partials))))
-        partial_embeds = torch.cat([self(batch) for batch in partials.chunk(n_chunks)], dim=0).cpu()
+        partial_embeds = torch.cat([self(batch) for batch in partials.chunk(n_chunks)], dim=0)
 
         # Reduce the partial embeds into full embeds and L2-normalize them
-        slices = np.concatenate(([0], np.cumsum(n_partials)))
+        slices = torch.tensor([0] + list(np.cumsum(n_partials)), device=partial_embeds.device)
```

Ruff (0.13.1)

Consider `[0, *list(np.cumsum(n_partials))]` instead of concatenation

Replace with `[0, *list(np.cumsum(n_partials))]`

(RUF005)